File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        light = 0 
        heavy = len(people) - 1
        
        people = sorted(people)
        count = 0 
        
        while light <= heavy:
            if people[light] + people[heavy] <= limit:
                light += 1
                
            heavy -= 1
            count += 1
            
            
        return count 
            
        

File: 917-boats-to-save-people/boats-to-save-people.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function(people, limit) {
    let light = 0;
    let heavy = people.length - 1;

    people.sort((a, b) => a - b);  // Sort numerically

    let count = 0;

    while (light <= heavy) {
        if (people[light] + people[heavy] <= limit) {
            light++;
        }
        heavy--;
        count++;
    }

    return count;
};